File: C_Programming/Practise_Programs/PP_169.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_169
//
//  Description       : This program toggles the 28th bit of a given
//                      integer number using the bitwise XOR operator.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 21/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef unsigned int UINT;

int main()
{
    UINT iNo = 0;
    UINT iMask = 0;

    printf("Enter Number : \n");
    scanf("%u", &iNo);

    iMask = 0x08000000;

    iNo = iNo ^ iMask;
    
    printf("Updated number : %u\n", iNo);

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      10
//
//  Sample Output :
//
//      Enter Number :
//      Updated number : 134217738
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_169.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_169
//
//  Description       : This program displays a butterfly pattern
//                      using stars and spaces.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 05/09/2026
//
//  Time Complexity   : O(n^2)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

class PP_169
{
    public static void main(String args[])
    {
        for(int i = 1; i <= 5; i++)  //First height
        {
            for(int j = 1; j <= i; j++)  //First star
            {
                System.out.print("*");
            }
            for(int j = 1; j <= (5-i) * 2; j++)  //space
            {
                System.out.print(" ");
            }
            for(int j = 1; j <= i; j++)  //Second Star
            {
                System.out.print("*");
            }
            System.out.println();
        }
        for(int i = 1; i <= 4; i++)  //height
        {
            for(int j = 1; j <= 5 - i; j++)  //First Star
            {
                System.out.print("*"); 
            }
            for(int j = 1; j <= i * 2; j++)  //space
            {
                System.out.print(" ");
            }
            for(int j = 1; j <= 5 - i; j++)  //Second Star
            {
                System.out.print("*");
            }
            System.out.println();
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//  *        *
//  **      **
//  ***    ***
//  ****  ****
//  **********
//  ****  ****
//  ***    ***
//  **      **
//  *        *
//
//////////////////////////////////////////////////////////////////